File: android/app/src/main/java/com/regta/core/REGTA.java
package com.regta.core;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

public class REGTA extends GTASA {

    private static native void initreGTA();

    @Override
    public void onCreate(Bundle savedInstanceState) {
        activity = this;

        try {
            initreGTA();
        } catch (UnsatisfiedLinkError e) {
            if (e.getMessage() != null) {
                Log.e("Error", e.getMessage());
            }
        }

        super.onCreate(savedInstanceState);
    }

    public static Activity activity;
}

File: android/app/src/main/java/com/regta/core/REGTA.kt
package com.gta.reversed

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity

class REGTA : GTASA() {

    private external fun initreGTA()

    override fun onCreate(savedInstanceState: Bundle?) {

        activity = this

        try {
            initreGTA()
        } catch (e: UnsatisfiedLinkError) {
            e.message?.let { Log.e("Error", it) }
        }

        super.onCreate(savedInstanceState)
    }

    companion object {
        @JvmStatic
        lateinit var activity: AppCompatActivity
    }
}